docs/flink-queries.md

```diff
@@ -277,6 +277,58 @@ DataStream<Row> stream = env.fromSource(source, WatermarkStrategy.noWatermarks()
     "Iceberg Source as Avro GenericRecord", new GenericRecordAvroTypeInfo(avroSchema));
 ```
 
+### Emitting watermarks
+Emitting watermarks from the source itself could be beneficial for several purposes, like harnessing the
+[Flink Watermark Alignment](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/datastream/event-time/generating_watermarks/#watermark-alignment)
+feature to prevent runaway readers, or providing triggers for [Flink windowing](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/datastream/operators/windows/).
+
+Enable watermark generation for an `IcebergSource` by setting the `watermarkColumn`.
+The supported column types are `timestamp`, `timestamptz` and `long`.
+Timestamp columns are automatically converted to milliseconds since the Java epoch of
+1970-01-01T00:00:00Z. Use `watermarkTimeUnit` to configure the conversion for long columns.
```

Comment:
> Use `watermarkTimeUnit` to configure the conversion for long columns.

Before this sentence, maybe elaborate a little more. 
```
Iceberg `timestamp` or `timestamptz` inherently contains the time precision. So there is no need to specify the time unit. But `long` type column doesn't contain time unit information. Use  `watermarkTimeUnit` to configure the conversion for long columns.
```

Reply:
Done